app/src/main/java/com/example/househomey/filter/ui/KeywordFilterFragment.java

```diff
@@ -1,22 +1,136 @@
 package com.example.househomey.filter.ui;
 
+import android.content.Context;
+import android.content.res.Resources;
 import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
 
+import androidx.core.content.ContextCompat;
+
+import com.example.househomey.R;
 import com.example.househomey.filter.model.FilterCallback;
+import com.example.househomey.filter.model.KeywordFilter;
+import com.google.android.material.chip.Chip;
+import com.google.android.material.chip.ChipGroup;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
+
+/**
+ * Fragment for applying "Keyword" filter criteria.
+ * @author Antonio Lech Martin-Ozimek
+ */
 public class KeywordFilterFragment extends FilterFragment {
+    private ChipGroup chipGroup = contentView.findViewById(R.id.chip_group_labels);
+    private ArrayList<String> chipTextVals = new ArrayList<>();
+    protected KeywordFilter keywordFilter;
+    Button addButton = contentView.findViewById(R.id.add_keyword_button);
+    EditText keyWords = contentView.findViewById(R.id.keyword_edit_text);
 
+    /**
+     * Constructs a new KeywordFilterFragment.
+     * @param title         The title of the "Keyword" filter dialog.
+     * @param contentView   The content view of the "Keyword" filter dialog.
+     * @param filterCallback The callback interface for handling filter changes.
+     * @param keywordFilter The previous filter instance
+     */
+    public KeywordFilterFragment(String title, View contentView, FilterCallback filterCallback, KeywordFilter keywordFilter) {
+        super(title, contentView, filterCallback);
+        this.keywordFilter = keywordFilter;
+        autoFillFilter(this.keywordFilter.getOgKeyWords());
+        addButton.setOnClickListener(new View.OnClickListener() {
```

Comment:
this is more of a nitpick, but can we change these `@Override OnClick`'s into lambda expressions to be a bit neater? (this one, and in the other fragment constructor). Click on the warning sign in top right corner of IDE
 
![image](https://github.com/CMPUT301F23T08/HouseHomey/assets/90405643/e119b2ba-b9b5-4121-a0eb-1e6e05684630)

Reply:
done.